Neos.Flow/Classes/Annotations/InjectConfiguration.php

```diff
@@ -49,21 +49,33 @@ final class InjectConfiguration
      *
      * @var string|null
      */
-    public $package;
+    public readonly ?string $package;
 
     /**
      * Type of Configuration (defaults to "Settings").
      *
      * @var string one of the ConfigurationManager::CONFIGURATION_TYPE_* constants
      */
-    public $type = ConfigurationManager::CONFIGURATION_TYPE_SETTINGS;
+    public readonly string $type;
 
     public function __construct(?string $path = null, ?string $package = null, ?string $type = null)
     {
         $this->path = $path;
         $this->package = $package;
-        if ($type !== null) {
-            $this->type = $type;
+        $this->type = $type ?? ConfigurationManager::CONFIGURATION_TYPE_SETTINGS;
+    }
```

Comment:
we should put the validation, that package cannot be used together with type if its a non setting.

previous error message

> The InjectConfiguration annotation for property "injectedViewsConfiguration" in class "Neos\Flow\Tests\Functional\ObjectManagement\Fixtures\ClassWithInjectedConfiguration" specifies a "package" key for configuration type "Views", but this is only supported for injection of "Settings".

doctrine will help us make a simple message readable:

> [Creation Error] An error occurred while instantiating the annotation @Flow\InjectConfiguration declared on property Neos\Flow\Tests\Functional\ObjectManagement\Fixtures\ClassWithInjectedConfiguration::$injectedViewsConfiguration: "Invalid usage of "package" with configuration type "Views". Using "package" is only valid for "Settings".".